universe/ValueRefs.cpp

```diff
@@ -304,37 +319,25 @@ std::string_view MeterToName(const MeterType meter) {
     return NAME_BY_METER[static_cast<std::underlying_type_t<MeterType>>(meter) + 1];
 }
 
-constexpr std::string_view PlanetTypeToStringConstexpr(PlanetType type) {
-    switch (type) {
-    case PlanetType::PT_SWAMP:     return "Swamp";
-    case PlanetType::PT_TOXIC:     return "Toxic";
-    case PlanetType::PT_INFERNO:   return "Inferno";
-    case PlanetType::PT_RADIATED:  return "Radiated";
-    case PlanetType::PT_BARREN:    return "Barren";
-    case PlanetType::PT_TUNDRA:    return "Tundra";
-    case PlanetType::PT_DESERT:    return "Desert";
-    case PlanetType::PT_TERRAN:    return "Terran";
-    case PlanetType::PT_OCEAN:     return "Ocean";
-    case PlanetType::PT_ASTEROIDS: return "Asteroids";
-    case PlanetType::PT_GASGIANT:  return "GasGiant";
-    default:                       return "?";
-    }
+std::string_view PlanetTypeToString(const PlanetType planet) {
+    // NOTE: INVALID_PLANET_TYPE (enum's -1 position) <= planet < NUM_PLANET_TYPES (enum's final position)
+    return NAME_BY_PLANET[static_cast<std::underlying_type_t<PlanetType>>(planet) + 1]; 
 }
 
-std::string_view PlanetTypeToString(PlanetType type)
-{ return PlanetTypeToStringConstexpr(type); }
-
 // @return the correct PlanetType enum for a user friendly planet type string (e.g. "Ocean"), else it returns PlanetType::INVALID_PLANET_TYPE
-PlanetType StringToPlanetType(std::string_view name) {
-    for (PlanetType pt = PlanetType::INVALID_PLANET_TYPE; pt < PlanetType::NUM_PLANET_TYPES;
-         pt = PlanetType(static_cast<std::underlying_type_t<PlanetType>>(pt) + 1))
-    {
-        if (PlanetTypeToStringConstexpr(pt) == name)
-            return pt;
+constexpr PlanetType StringToPlanetType(const std::string_view name) {
```

Comment:
I'm not sure if `constexpr` here is the way to go, but the new function does 1 less lookup.